plugins/device/cmd/nvidia/fingerprint.go

```diff
@@ -0,0 +1,172 @@
+package nvidia
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/nomad/plugins/device"
+	nvml "github.com/hashicorp/nomad/plugins/device/cmd/nvidia/nvml_client"
+)
+
+const (
+	// Attribute names for reporting Fingerprint output
+	MemoryMiBAttr     = "memory_mib"
+	DriverVersionAttr = "driver_version"
+)
+
+const (
+	baseTen = 10
+)
+
+// fingerprint is the long running goroutine that detects hardware
+func (d *NvidiaDevice) fingerprint(ctx context.Context, devices chan<- *device.FingerprintResponse) {
+	defer close(devices)
+
+	if d.nvmlCli.CheckDriverInitialization() != nil {
```

Comment:
If the NewNvmlCli returns an error you can store that and then this and fingerprint can use that to exit early when the nvml client isn't working